internal/util/password/scramsha256_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package password
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/xdg-go/scram"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+// scramSHA256TestCase represents a test case for SCRAM-SHA-256 authentication.
+//
+//nolint:vet // for readability
+type scramSHA256TestCase struct {
+	params   scramSHA256Params
+	username string
+	password string
+	salt     []byte
+
+	want *types.Document
+	err  string
+}
+
+var scramSHA256TestCases = []scramSHA256TestCase{
```

Comment:
The exported `SCRAMSHA256Hash()` uses fixed length. Why do we test many combination len and count of `scramSHA256Params`?

Reply:
The idea is to simplify migration from MongoDB by supporting their parameters. So we support _importing_ `system.users` collection from MongoDB with different parameters, but use fixed parameters for _creating_ new users. Part of it is not there yet.